src/auth/CheckAuth.tsx

```diff
@@ -0,0 +1,30 @@
+import { useEffect } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { setAuthHeader } from '../services/axios';
+import { useAuth } from './Context';
+
+const CheckAuth = ({ children }: { children: JSX.Element }) => {
+    const { user } = useAuth();
+    const { pathname } = useLocation();
+    const navigate = useNavigate();
+
+    useEffect(() => {
+        if (user) {
+            setAuthHeader(user);
+            if (pathname === '/' || pathname.includes('login')) {
+                navigate('/dashboard', { replace: true });
+            } else {
+                navigate(pathname, { replace: false });
+            }
+        }
```

Comment:
As long as there is a user we should set the auth header for axios. This includes if they log out and user is null.

Also - if they are already logged in and trying to go to either of the "log in paths" then we should send 'em to the main page.